ci-templates/gitlab/image-builder.yml

```
@@ -420,3 +423,24 @@ t4c/client/ease/remote/debug:
       --build-arg BASE_IMAGE=$BASE_IMAGE \
       ease/debug
     - *push
+
+jupyter:
+  stage: build
+  needs:
+    - job: base
+      optional: true
+  rules:
+    - if: '$JUPYTER == "1"'
+      when: always
+  variables:
+    BASE_IMAGE: $DOCKER_REGISTRY/base2
+    IMAGE: $DOCKER_REGISTRY/jupyter-notebook
+  script:
+    - *prepare
+    - *docker
+    - |
+      docker build $DOCKER_BUILD_ARGS \
+      -t $IMAGE:python-3.11-$CI_COMMIT_REF_NAME \
+      --build-arg BASE_IMAGE=$BASE_IMAGE \
+      jupyter-notebook
+    - docker push $IMAGE:python-3.11-$CI_COMMIT_REF_NAME
```

```suggestion
    - |
      docker build $DOCKER_BUILD_ARGS \
      -t $DOCKER_REGISTRY/jupyter-notebook/$DOCKER_TAG \
      --build-arg BASE_IMAGE=$BASE_IMAGE \
      jupyter-notebook
    - *push
```
By doing it like this (you need to apply the `before_script` comment for this), we keep the same structure as for the other jobs we have defined here. Whats your opinion about this?